mlos_bench/mlos_bench/run.py

```diff
@@ -101,50 +94,26 @@ def _optimize(*,
 
         _LOG.info("Experiment: %s Env: %s Optimizer: %s", exp, env, opt)
 
+        last_trial_id = -1
         if opt_context.supports_preload:
-            # Load (tunable values, benchmark scores) to warm-up the optimizer.
-            # `.load()` returns data from ALL merged-in experiments and attempts
-            # to impute the missing tunable values.
-            (configs, scores, status) = exp.load()
-            opt_context.bulk_register(configs, scores, status)
-            # Complete any pending trials.
-            for trial in exp.pending_trials(datetime.utcnow(), running=True):
-                _run(env_context, opt_context, trial, global_config)
+            # Complete trials that are pending or in-progress.
+            _scheduler(exp, env_context, global_config, running=True)
+            # Load past trials data into the optimizer
+            last_trial_id = _optimizer(exp, opt_context, is_warm_up=True)
```

Comment:
According to the `is_warm_up` flag, this won't increment the `_iter` counter, right?

Reply:
no, it shouldn't - this is a warm-up phase, and we don't want to increment the counter here. at least, this is the logic that we've had before.

Reply:
Yeah, just means that whenever you resume, the max_iterations is constantly moving forward.
For instance, then if you run 90 iterations, kill the script, and then start it again until completion, you will run for 190 total iterations, rather than just 10 more.
Not sure which is more correct atm.
Maybe we can take that up more formally in #427